src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -42,345 +128,186 @@ object WdlNamespace {
-
-  private def processWdlType(ast: AstNode): WdlType = {
-    ast match {
-      case t: Terminal =>
-        t.getSourceString match {
-          case WdlFileType.toWdlString => WdlFileType
-          case WdlStringType.toWdlString => WdlStringType
-          case WdlIntegerType.toWdlString => WdlIntegerType
-          case WdlFloatType.toWdlString => WdlFloatType
-          case WdlBooleanType.toWdlString => WdlBooleanType
-          case WdlObjectType.toWdlString => WdlObjectType
-        }
-      case null => WdlStringType
-      case _ => throw new UnsupportedOperationException("Implement this later for compound types")
+    // FIXME: Here's where I'd toSet stuff after duplications are detected
+    ast.findAsts(AstNodeName.Workflow) match {
+      case Nil => NamespaceWithoutWorkflow(namespace, imports, namespaces, tasks, importedWorkflows, terminalMap, ast)
+      case Seq(x) => NamespaceWithWorkflow(ast, x, namespace, imports, namespaces, tasks, terminalMap, importedWorkflows, wdlSyntaxErrorFormatter)
+      case doh => throw new SyntaxError(wdlSyntaxErrorFormatter.tooManyWorkflows(doh.asJava))
     }
   }
 
+
   /**
-   * Confirm all required inputs are present and attempt to coerce raw inputs to `WdlValue`s.
-   * This can fail if required raw inputs are missing or if the values for a specified raw input
-   * cannot be coerced to the target type of the input as specified in the namespace.
+   * Given a name, a collection of WdlNamespaces and a collection of Tasks will attempt to find a Task
+   * with that name within the WdlNamespaces
    */
-  def coerceRawInputs(rawInputs: WorkflowRawInputs): Try[WorkflowCoercedInputs] = {
-
-    def coerceRawInput(fqn: FullyQualifiedName, wdlType: WdlType): Try[WdlValue] = fqn match {
-      case _ if rawInputs.contains(fqn) =>
-        val rawInput = rawInputs.get(fqn).get
-        wdlType.coerceRawValue(rawInput).recoverWith {
-          case e => Failure(new UnsatisfiedInputsException(s"Failed to coerce input $fqn value $rawInput of ${rawInput.getClass} to $wdlType."))
-        }
-      case _ => Failure(new UnsatisfiedInputsException(s"Required workflow input '$fqn' not specified."))
-    }
-
-    val tryCoercedValues = workflows.head.inputs.map { case (fqn, wdlType) =>
-      fqn -> coerceRawInput(fqn, wdlType)
-    }
-
-    val (successes, failures) = tryCoercedValues.partition { case (_, tryValue) => tryValue.isSuccess }
-    if (failures.isEmpty) {
-      Try(successes.map { case (key, tryValue) => key -> tryValue.get })
-    } else {
-      val message = failures.values.collect { case f: Failure[_] => f.exception.getMessage }.mkString("\n")
-      Failure(new UnsatisfiedInputsException(message))
-    }
+  def findTask(name: String, namespaces: Seq[WdlNamespace], tasks: Seq[Task]): Option[Task] = {
+    if (name.contains(".")) {
+      val parts = name.split("\\.", 2)
+      namespaces find {_.importedAs == Some(parts(0))} flatMap {x => findTask(parts(1), x.namespaces, x.tasks)}
```

Comment:
Agreed. This is another spot where the short answer is "it's what was there before", although the else branch is different I believe.